src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -72,7 +85,42 @@ export class DotEditContentComponent implements OnInit, OnDestroy {
         public dotEditContentHtmlService: DotEditContentHtmlService,
         public dotLoadingIndicatorService: DotLoadingIndicatorService,
         public sanitizer: DomSanitizer
-    ) {}
+    ) {
+        if (!this.customEventsHandler) {
+            this.customEventsHandler = {
+                'load-edit-mode-page': (pageRendered: DotRenderedPage) => {
+                    const dotRenderedPageState = new DotRenderedPageState(this.pageState.user, pageRendered);
+
+                    if (this.route.snapshot.queryParams.url === pageRendered.page.pageURI) {
+                        this.setPageState(dotRenderedPageState);
+                    } else {
+                        this.dotEditPageDataService.set(dotRenderedPageState);
+                        this.dotRouterService.goToEditPage(pageRendered.page.pageURI);
+                    }
+                },
+                'in-iframe': () => {
+                    this.reload();
+                },
+                'reorder-menu': (reorderMenu: DotReorderMenu) => {
+                    this.reorderMenuUrl = [
```

Comment:
It is worthy to build this url here? why don't just build it in core and send it in the event?